### ISSUE TITLE
Nothing returns when used attributes with JSON field in table on MariaDB dialect

### ISSUE THREAD

**Issue by hoanghog** [sequelize/sequelize#13027]
### What are you doing?
Here is the link with the problem https://github.com/roychri/sequelize_debug_2021-02-11

Trying to query a table with JSON type field with attribute projection, but if I don't state JSON field in attribute it will return nothing.
### What do you expect to happen?
Wanted to retrieve data with attribute projection.

### What is actually happening?
Returned nothing

### Additional context
As described on git page I linked, if you remove the JSON field from model, it will works normally.
### Environment

- Sequelize version: 5.21.7 // Doesn't really matter
- Node.js version: 14.4.0
- Operating System: Windows

### How does this problem relate to dialects?

- [ ] I think this problem happens regardless of the dialect.
- [X] I think this problem happens only for the following dialect(s): MariaDB
- [ ] I don't know, I was using PUT-YOUR-DIALECT-HERE, with connector library version XXX and database version XXX

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I don't know how to start, I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [X] No, I don't have the time and I wouldn't even know how to start.

**Comment by alexjavier15** [comment]
> ### What are you doing?
> Here is the link with the problem https://github.com/roychri/sequelize_debug_2021-02-11
> 
> Trying to query a table with JSON type field with attribute projection, but if I don't state JSON field in attribute it will return nothing.
> 
> ### What do you expect to happen?
> Wanted to retrieve data with attribute projection.
> 
> ### What is actually happening?
> Returned nothing
> 
> ### Additional context
> As described on git page I linked, if you remove the JSON field from model, it will works normally.
> 
> ### Environment
> * Sequelize version: 5.21.7 // Doesn't really matter
> * Node.js version: 14.4.0
> * Operating System: Windows
> 
> ### How does this problem relate to dialects?
> * [ ]  I think this problem happens regardless of the dialect.
> * [x]  I think this problem happens only for the following dialect(s): MariaDB
> * [ ]  I don't know, I was using PUT-YOUR-DIALECT-HERE, with connector library version XXX and database version XXX
> 
> ### Would you be willing to resolve this issue by submitting a Pull Request?
> * [ ]  Yes, I have the time and I know how to start.
> * [ ]  Yes, I have the time but I don't know how to start, I would need guidance.
> * [ ]  No, I don't have the time, although I believe I could do it if I had the time...
> * [x]  No, I don't have the time and I wouldn't even know how to start.

I'm facing the same problem on mariadb. If I add a JSON field to the list of exluded attributes,  the query returns nothing.

I debugged it a bit. As far I can see, the results returned by mariadb query connection,  always return null for the JSON attribute, even if its not part of the Select statement od the SQL query. Somehow this break the query handler (_groupJoinData function to be more specific.

https://github.com/sequelize/sequelize/blob/5b16b32259f0599a6af2d1eb625622da9054265e/lib/dialects/mariadb/query.js#L34-L50

**Comment by alexjavier15** [comment]
I'm facing the same problem on mariadb. If I add a JSON field to the list of exluded attributes, the query returns nothing.

I digged a bit. As far I can see, the results returned by mariadb query connection, always return null for the JSON attribute, even if its not part of the Select statement od the SQL query. Somehow this break the query handler (_groupJoinData function to be more specific. This is caused by the handleJsonSelectQuery.

https://github.com/sequelize/sequelize/blob/master/src/dialects/mariadb/query.js#L180-L185

I changed the logic to skip setting null to the JSON attribute by default in my project and worked just fine.

Check the workaround in  #12583
